### ISSUE TITLE
Add Azure MSSQL Database

### ISSUE THREAD

**Issue by aliscott** [infracost/infracost#569]
Pricing page: https://azure.microsoft.com/en-gb/pricing/details/sql-database/single/

`azurerm_mssql_database`

* `Compute (provisioned, GP_Gen5_2)` or `Compute (serverless, GP_S_Gen5_1)` - unit is hours.
  * The instance type can be taken from `sku_name`. If this is `ElasticPool` then this resource has no cost since it will be handled by the `azurerm_sql_elastic_pool` resource.
  * `zone_redundant`  is used for both Compute and Storage pricing
  * If `license_type` is `BasePrice` we should use the Hybrid Benefit (same as dev/test) pricing.
  * For serverless, the unit should be `vCore-hours` and should be usage based, the usage key should be `vcore_hours`.
* `Storage` - unit is GB-months. The quantity should be based on `max_size_gb`
* `Read replicas` - unit is hours. This is only used for Hyperscale DBs. Quantity is based on `read_replica_count`.
* `Long-term retention` - unit is GB-months. Usage-based, usage key should be `monthly_long_term_retention_storage_gb`

Unsupported:
* Threat detection

**Comment by ekifel** [comment]
@alikhajeh1, hey, I want to work on it!

**Comment by alikhajeh1** [comment]
@ekifel can you please take a look at these warnings that are happening on the latest master:
```
=== RUN   TestMSSQLDatabase
time="2021-05-04T12:33:10Z" level=warning msg="Multiple products found for azurerm_mssql_database.business_critical_gen Compute (provisioned, BC_Gen5_8), using the first product"
time="2021-05-04T12:33:10Z" level=warning msg="Multiple products found for azurerm_mssql_database.business_critical_m Compute (provisioned, BC_M_8), using the first product"
time="2021-05-04T12:33:10Z" level=warning msg="Multiple products found for azurerm_mssql_database.general_purpose_gen Compute (provisioned, GP_Gen5_4), using the first product"
--- PASS: TestMSSQLDatabase (11.75s)
=== RUN   TestMSSQLDatabase_HS_with_replicas
--- PASS: TestMSSQLDatabase_HS_with_replicas (11.06s)
=== RUN   TestMSSQLDatabase_with_license
time="2021-05-04T12:33:32Z" level=warning msg="Multiple products found for azurerm_mssql_database.general_purpose_gen Compute (provisioned, GP_Gen5_4), using the first product"
--- PASS: TestMSSQLDatabase_with_license (10.89s)
=== RUN   TestMSSQLDatabase_zone_redundant
time="2021-05-04T12:33:43Z" level=warning msg="Multiple products found for azurerm_mssql_database.general_purpose_gen Compute (provisioned, GP_Gen5_4), using the first product"
--- PASS: TestMSSQLDatabase_zone_redundant (10.89s)
=== RUN   TestMSSQLDatabase_serverless
--- PASS: TestMSSQLDatabase_serverless (11.34s)
=== RUN   TestMSSQLDatabase_LTR
time="2021-05-04T12:34:06Z" level=warning msg="Multiple products found for azurerm_mssql_database.my_db Compute (provisioned, GP_Gen5_4), using the first product"
--- PASS: TestMSSQLDatabase_LTR (11.41s)
```

**Comment by alikhajeh1** [comment]
Released in v0.8.6
